cds/modules/deposit/api.py

```diff
@@ -993,9 +989,8 @@ def discard(self, pid=None):
         """Discard a video."""
         video_old_ref = self.ref
         video_discarded = super(Video, self).discard(pid=pid)
-        video_discarded.project._update_videos(
-            [video_old_ref], [video_discarded.ref]
-        )
+        video_discarded.project._update_videos([video_old_ref], [video_discarded.ref])
+        video_discarded.project.commit()
```

Comment:
out of curiosity: why we need a commit?

Reply:
because the project was needs to be committed to add to the session the changes after the removal of videos. We do the same here: https://github.com/CERNDocumentServer/cds-videos/blob/fa19983ea90f1037a4c044b0d14d0927334b6043/cds/modules/deposit/api.py#L949